### PR TITLE
Gracefully handle creating milestone with existing name

### PR DESCRIPTION
**Pull request by misiekhardcore**
related to  #https://github.com/bpmn-io/internal-docs/issues/1052

### Proposed Changes

 

Check if milestone exists before trying to create it to prevent the action from exiting with an error when trying to create a milestone which already exists (e.g. https://github.com/camunda/camunda-modeler/actions/runs/11625285084/job/32375114566)

Previous approach (done in #4639 ) turned out to not work as expected in GHA environment

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [x] __Visual demo__ attached
* [ ] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: .github/workflows/CREATE_MILESTONE.yml [hunk 1]
```diff
@@ -12,32 +12,35 @@ jobs:
...
         id: getTitle
-        env: 
+        env:
           TITLE: ${{github.event.milestone.title}}
         run: |
           MILESTONE_NUMBER=${TITLE:1}
           INCREMENTED_NUMBER=$((MILESTONE_NUMBER + 1))
           echo "MILESTONE_NAME=M$INCREMENTED_NUMBER" >> $GITHUB_OUTPUT
+      - name: Check if Milestone already exists
+        id: checkIfMilestoneExists
```
Comment (nikku) [inline .github/workflows/CREATE_MILESTONE.yml:22]: Consider our discussions on steps naming in https://github.com/camunda/camunda-modeler/pull/4650 adjust your changes accordingly.

**Comment by nikku** [comment]
Fun fact: I thought about this approach as I reviewed your previous contribution. I ended up ditching it, as both have their merits. I do think this is cleaner and more robust though, so great to see we're taking this route now.

**Review by nikku — APPROVED** [review]
Let's test this in real life to verify it works.

**Comment by misiekhardcore** [comment]
I updated the naming according to our new convention and I am merging it as soon as CI passes

**Pull request merged**